src/app/modules/account/services/msp-account-data.service.ts

```diff
@@ -258,7 +258,7 @@ export class MspAccountMaintenanceDataService {
     output.firstName = dto.firstName;
     output.middleName = dto.middleName;
     output.lastName = dto.lastName;
-    output.dob = dto.dob;
+    output.dob = dto.dob ? new Date(dto.dob) : undefined;
```

Comment:
Converting strings to dates so they can be handled. It's repetitive but I wanted to change as little as possible due to the time constraint. Previously they were being taken directly out of localstorage (as strings) and this would cause errors because Date types are expected